src/BuiltInTools/BrowserRefresh/WebSocketScriptInjection.js

```diff
@@ -187,9 +194,37 @@ setTimeout(async function () {
     connection.send(new Uint8Array([0]).buffer);
   }
 
+  async function getSecret(serverKeyString) {
+    if (!serverKeyString || !window.crypto || !window.crypto.subtle) {
+      return null;
+    }
+
+    const secretBytes = window.crypto.getRandomValues(new Uint8Array(32)); // 32-bytes of entropy
+
+    // Based on https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/importKey#subjectpublickeyinfo_import
+    const binaryServerKey = str2ab(atob(serverKeyString));
+    const serverKey = await window.crypto.subtle.importKey('spki', binaryServerKey, { name: "RSA-OAEP", hash: "SHA-256" }, false, ['encrypt']);
+    const encrypted = await window.crypto.subtle.encrypt({ name: 'RSA-OAEP' }, serverKey, secretBytes);
+    return {
+      encryptedSharedSecret: btoa(String.fromCharCode(...new Uint8Array(encrypted))),
+      encodedSharedSecret: btoa(String.fromCharCode(...secretBytes)),
+    };
+
+    function str2ab(str) {
+      const buf = new ArrayBuffer(str.length);
+      const bufView = new Uint8Array(buf);
+      for (let i = 0, strLen = str.length; i < strLen; i++) {
+        bufView[i] = str.charCodeAt(i);
+      }
+      return buf;
+    }
+  }
+
   function getWebSocket(url) {
     return new Promise((resolve, reject) => {
-      const webSocket = new WebSocket(url);
+      const encryptedSecret = sharedSecret && sharedSecret.encryptedSharedSecret;
+      const protocol = encryptedSecret ? encodeURIComponent(encryptedSecret) : null;
+      const webSocket = new WebSocket(url, protocol);
```

Comment:
@BrennanConroy updated.

Reply:
Are there end-to-end tests? Looks like nothing failed with the query string

Reply:
It's going to be tricky to add browser based automated tests to this repo. We don't have a very good track record of making Selenium / Playwrite tests reliable and I wouldn't want them to destabilize the SDK repo. For the interim, my plan is to add some automated tests around the watch inner loop and see where that takes us.